src/twisted/newsfragments/7530.bugfix

```diff
@@ -0,0 +1 @@
+Allow for key and cert .pem files that do not have a a trailing newline
```

Comment:
My understanding is that the issue was not about lacking a newline at the end of a standalone file per se but about how we go about merging two files into one when their paths are provided in an `ssl:` endpoint.  Previously we created a combined end/start line like this.

    -----END PRIVATE KEY----------BEGIN CERTIFICATE-----

I added a test loading just the `cert.pem.no_trailing_newline` file via `ssl.Certificate.loadPEM()` and it worked fine.  \*crosses fingers that the build servers say the same thing\*

Did I miss the issue you are pointing out?  If someone provides a (I presume) malformed file with the line above instead of two separate lines wouldn't that be their problem to resolve?  We are only responsible for not creating an invalid file by not creating the bogus line above.  That's what the patch does.

If I have to I suppose I can dive into the RFCs for PEM files.